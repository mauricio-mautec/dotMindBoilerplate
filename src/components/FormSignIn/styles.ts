import styled, { css } from 'styled-components'
import { lighten } from 'polished'

export const ForgotPassword = styled.a`
  ${({ theme }) => css`
    display: block;
    text-decoration: none;
    text-align: right;
    font-size: ${theme.font.sizes.small};
    color: ${theme.colors.black};

    &:hover {
      color: ${lighten(0.3, theme.colors.black)};
    }
  `}
`
