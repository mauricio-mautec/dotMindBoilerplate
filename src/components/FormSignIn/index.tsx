import Link from 'next/link'
import { Lock, Email } from '@styled-icons/material-outlined'

import { FormWrapper, FormLink } from 'components/Form'
import TextField from 'components/TextField'
import Button from 'components/Button'

import * as S from './styles'

export type FormSignInProps = {
  labelButton: string
  labelLinkPassword: string
  labelLinkAccount: string
}
const FormSignIn = ({
  labelButton = 'Sign in now',
  labelLinkPassword = 'Forgot your password?',
  labelLinkAccount = 'Do not have an account?'
}: FormSignInProps) => (
  <FormWrapper>
    <TextField
      label="email"
      placeholder="Email"
      type="email"
      icon={<Email />}
    />
    <TextField
      label="password"
      placeholder="Password"
      type="password"
      icon={<Lock />}
    />

    <S.ForgotPassword href="#">{labelLinkPassword}</S.ForgotPassword>

    <Button size="large" fullWidth>
      {labelButton}
    </Button>

    <FormLink>
      {labelLinkAccount}{' '}
      <Link href="/sign-up">
        <a>Sign up</a>
      </Link>
    </FormLink>
  </FormWrapper>
)

export default FormSignIn
