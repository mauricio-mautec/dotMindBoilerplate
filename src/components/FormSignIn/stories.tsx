import { ComponentStory, ComponentMeta } from '@storybook/react'

import FormSignIn from '.'

export default {
  title: 'Form/FormSignIn',
  component: FormSignIn
} as ComponentMeta<typeof FormSignIn>

const Template: ComponentStory<typeof FormSignIn> = (args) => (
  <div style={{ width: 300, margin: 'auto' }}>
    <FormSignIn {...args} />
  </div>
)

export const Basic = Template.bind({})

Basic.args = {
  labelLinkAccount: 'Do not have an account?',
  labelButton: 'Sign in now',
  labelLinkPassword: 'Forgot your password?'
}
