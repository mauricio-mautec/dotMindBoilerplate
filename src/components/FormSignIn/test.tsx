import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/tests/helpers'

import FormSignIn from '.'

describe('<FormSignIn />', () => {
  it('Renders with placeholder', () => {
    const { container } = renderWithTheme(
      <FormSignIn
        labelButton="Sign in now"
        labelLinkPassword="Forgot your password?"
        labelLinkAccount="Do not have an account?"
      />
    )

    expect(screen.getByPlaceholderText('Email')).toBeInTheDocument()
    expect(screen.getByPlaceholderText('Password')).toBeInTheDocument()
    expect(
      screen.getByRole('link', { name: /forgot your password/i })
    ).toBeInTheDocument()
    expect(screen.getByRole('button', { name: /sign in now/i }))
    expect(screen.getByRole('link', { name: /Sign up/i })).toBeInTheDocument()
    expect(screen.getByText(/do not have an account?/i)).toBeInTheDocument()

    expect(container).toMatchSnapshot()
  })
})
