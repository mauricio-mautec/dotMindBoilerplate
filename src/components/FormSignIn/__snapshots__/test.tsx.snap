// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormSignIn /> Renders with placeholder 1`] = `
.c5 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c2 {
  font-size: 1.4rem;
  color: #030517;
  cursor: pointer;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background: #F2F2F2;
  border-radius: 0.2rem;
  padding: 0 1.6rem;
  border: 0.2rem solid;
  border-color: #F2F2F2;
}

.c3:focus-within {
  box-shadow: 0 0 0.5rem #003399;
}

.c6 {
  color: #030517;
  font-family: Poppins,-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Open Sans','Helvetica Neue',sans-serif;
  font-size: 1.6rem;
  padding: 0.8rem 0;
  padding-left: 1.6rem;
  background: transparent;
  border: 0;
  outline: none;
  width: 100%;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 2.2rem;
  color: #8D8D8D;
  -webkit-order: 0;
  -ms-flex-order: 0;
  order: 0;
}

.c4 > svg {
  width: 100%;
}

.c9 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background: linear-gradient(180deg,#ff5f5f 0%,#f062c0 50%);
  color: #FAFAFA;
  border: 0;
  cursor: pointer;
  border-radius: 0.4rem;
  padding: 0.8rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  height: 5rem;
  font-size: 1.6rem;
  padding: 0.8rem 4.8rem;
  width: 100%;
}

.c9:hover {
  background: linear-gradient(180deg,#e35565 0%,#d958a6 50%);
}

.c0 .c1 {
  margin: 0.8rem 0;
}

.c0 .c8 {
  margin: 3.2rem auto 1.6rem;
}

.c10 {
  font-size: 1.4rem;
  color: #030517;
  text-align: center;
}

.c10 a {
  color: #E2306C;
  -webkit-text-decoration: none;
  text-decoration: none;
  border-bottom: 0.1rem solid #E2306C;
  -webkit-transition: color,border,0.1s ease-in-out;
  transition: color,border,0.1s ease-in-out;
}

.c10 a:hover {
  color: #c41b54;
  border-bottom: 0.1rem solid #c41b54;
}

.c7 {
  display: block;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-align: right;
  font-size: 1.4rem;
  color: #030517;
}

.c7:hover {
  color: #15229e;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1 "
    >
      <label
        class="c2"
        for=""
      >
        email
      </label>
      <div
        class="c3"
      >
        <div
          class="c4"
        >
          <svg
            aria-hidden="true"
            class="c5"
            fill="currentColor"
            focusable="false"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0V0z"
              fill="none"
            />
            <path
              d="M22 6c0-1.1-.9-2-2-2H4c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6zm-2 0-8 5-8-5h16zm0 12H4V8l8 5 8-5v10z"
            />
          </svg>
        </div>
        <input
          class="c6"
          id=""
          placeholder="Email"
          type="email"
          value=""
        />
      </div>
    </div>
    <div
      class="c1 "
    >
      <label
        class="c2"
        for=""
      >
        password
      </label>
      <div
        class="c3"
      >
        <div
          class="c4"
        >
          <svg
            aria-hidden="true"
            class="c5"
            fill="currentColor"
            focusable="false"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              fill="none"
            >
              <path
                d="M0 0h24v24H0V0z"
              />
              <path
                d="M0 0h24v24H0V0z"
                opacity="0.87"
              />
            </g>
            <path
              d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zM9 6c0-1.66 1.34-3 3-3s3 1.34 3 3v2H9V6zm9 14H6V10h12v10zm-6-3c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2z"
            />
          </svg>
        </div>
        <input
          class="c6"
          id=""
          placeholder="Password"
          type="password"
          value=""
        />
      </div>
    </div>
    <a
      class="c7"
      href="#"
    >
      Forgot your password?
    </a>
    <button
      class="c8 c9"
    >
      <span>
        Sign in now
      </span>
    </button>
    <div
      class="c10"
    >
      Do not have an account?
       
      <a
        href="/sign-up"
      >
        Sign up
      </a>
    </div>
  </div>
</div>
`;
